lib/stripe/stripe_client.rb

```diff
@@ -382,12 +414,15 @@ def self.maybe_gc_connection_managers
 
       pruned_thread_contexts = []
       @thread_contexts_with_connection_managers.each do |thread_context|
-        connection_manager = thread_context.default_connection_manager
-        next if connection_manager.last_used > last_used_threshold
-
-        connection_manager.clear
-        thread_context.default_connection_manager = nil
-        pruned_thread_contexts << thread_context
+        thread_context
+          .default_connection_managers
+          .each do |_, connection_manager|
+            next if connection_manager.last_used > last_used_threshold
+
+            connection_manager.clear
+            thread_context.reset_connection_managers
+            pruned_thread_contexts << thread_context
```

Comment:
And similarly, I wonder if we should only prune the thread context once all connect managers in it have confirmed to not be in use. This might be easiest to do by just introducing a second loop that runs after we prune all eligible connection managers.

Reply:
Sounds reasonable to me. I'll get that update added 👍

Reply:
Excellent! Thx again.